File: content/problems/0697-degree-of-an-array/solutions/count-first-last/solution.cpp
class Solution {
    public: int findShortestSubArray(vector<int>& nums){
        unordered_map<int,int> first,count;
        int degree=0,length=0;
        for(int i=0;i<(int)nums.size();++i){
            int value=nums[i];
            if(!first.count(value))first[value]=i;
            int frequency=++count[value],span=i-first[value]+1;
            if(frequency>degree){
                degree=frequency;
                length=span;
            }else if(frequency==degree)length=min(length,span);
        }
        return length;
    }
};

File: content/problems/0697-degree-of-an-array/solutions/count-first-last/Solution.kt
class Solution {
    fun findShortestSubArray(nums:IntArray):Int{
        val first=hashMapOf<Int,Int>();
        val count=hashMapOf<Int,Int>();
        var degree=0;
        var length=0;
        for(i in nums.indices){
            val value=nums[i];
            first.putIfAbsent(value,i);
            val frequency=(count[value]?:0)+1;
            count[value]=frequency;
            val span=i-first[value]!!+1;
            if(frequency>degree){
                degree=frequency;
                length=span
            }else if(frequency==degree)length=minOf(length,span)
        };
        return length
    }
}
